src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -963,6 +1043,44 @@ impl Coordinator {
         Ok(Self::send_immediate_rows(rows))
     }
 
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn peek_stage_copy_to_finish(
+        &mut self,
+        ctx: &mut ExecuteContext,
+        PeekStageCopyToFinish {
+            validity: _,
+            optimizer,
+            global_lir_plan,
+            copy_to_context,
+        }: PeekStageCopyToFinish,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        let session = ctx.session_mut();
+        let (peek_plan, df_meta) = global_lir_plan.unapply();
+        let PeekPlan::SlowPath(PeekDataflowPlan {
+            desc: dataflow,
+            id: _,
+            ..
+        }) = peek_plan
+        else {
+            // While optimizing we had made sure to always go for the slow path in case of copy to
+            unreachable!()
+        };
+
+        self.emit_optimizer_notices(&*session, &df_meta.optimizer_notices);
+
+        // Ship the dataflow
+        self.controller
+            .active_compute()
+            .create_dataflow(optimizer.cluster_id(), dataflow)
+            .unwrap_or_terminate("cannot fail to create dataflows");
+
+        // TODO(mouli): implement!
```

Comment:
What other things (besides shipping the dataflow) do you intend to do here?

Is the `session` meant to block until (a) the `COPY TO` is completed or (b) do we want to do this asynchronously.

If (a) I suggest to look at `subscribe_finish` for reference. If (b) maybe `create_materialized_view_finish` is a better starting point.

Reply:
The session is meant to block. Yup, IMO it should look similar to a subscribe. I will do that in a follow up PR to limit the scope in the current one.